datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeRecoveryFlowEventChainFactory.java

```diff
@@ -0,0 +1,68 @@
+package com.sequenceiq.datalake.flow.chain;
+
+import static com.sequenceiq.datalake.flow.delete.SdxDeleteEvent.SDX_DELETE_EVENT;
+import static com.sequenceiq.datalake.flow.detach.SdxDetachEvent.SDX_DETACH_RECOVERY_EVENT;
+import static com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent.SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+import static com.sequenceiq.datalake.flow.start.SdxStartEvent.SDX_START_EVENT;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.flow.delete.event.SdxDeleteStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.SdxStartDetachRecoveryEvent;
+import com.sequenceiq.datalake.flow.start.event.SdxStartStartEvent;
+import com.sequenceiq.flow.core.chain.FlowEventChainFactory;
+import com.sequenceiq.flow.core.chain.config.FlowTriggerEventQueue;
+
+@Component
+public class DatalakeResizeRecoveryFlowEventChainFactory implements FlowEventChainFactory<DatalakeResizeRecoveryFlowChainStartEvent> {
+    @Override
+    public String initEvent() {
+        return SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+    }
+
+    @Override
+    public FlowTriggerEventQueue createFlowTriggerEventQueue(DatalakeResizeRecoveryFlowChainStartEvent event) {
+        Queue<Selectable> flowChain = new ConcurrentLinkedQueue<>();
+
+        // Forced deletion of the resized datalake.
+        boolean firstEventAdded = false;
+        if (event.getNewCluster().getDeleted() == null) {
+            flowChain.add(new SdxDeleteStartEvent(
+                    SDX_DELETE_EVENT.event(), event.getNewCluster().getId(), event.getUserId(), true, event.accepted()
+            ));
+            firstEventAdded = true;
+        }
+
+        // Reattach of the old datalake.
+        if (event.getOldCluster().isDetached()) {
+            if (firstEventAdded) {
+                flowChain.add(new SdxStartDetachRecoveryEvent(
+                        SDX_DETACH_RECOVERY_EVENT.event(), event.getOldCluster().getId(), event.getUserId()
+                ));
+            } else {
+                flowChain.add(new SdxStartDetachRecoveryEvent(
+                        SDX_DETACH_RECOVERY_EVENT.event(), event.getOldCluster().getId(), event.getUserId(), event.accepted()
```

Comment:
How is it different from what is done in line 44?

Reply:
The `event.accepted` part is needed for the first event in a flow chain otherwise you get some errors and even a UI popup warning. 
The flow chain still runs but I figured we would not want warnings or errors to appear from the user's perspective.

Reply:
@foldik  I see that makes sense, thank you for providing the example. I'll clean this up.

Reply:
Done.